datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.java

```diff
@@ -29,36 +30,58 @@ public class GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter {
 
     static final String VALIDATION_FAILED_STATE = "VALIDATION_FAILED";
 
+    static final String SUCCESSFUL_STATUS = "SUCCESSFUL";
+
     private static final Set<String> FAILED_STATES = Set.of(FAILED_STATE, VALIDATION_FAILED_STATE);
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.class);
 
     public DatalakeBackupStatusResponse convert(BackupDatalakeResponse response) {
         return new DatalakeBackupStatusResponse(response.getBackupId(),
-                DatalakeBackupStatusResponse.State.valueOf(response.getOverallState()),
-                Optional.ofNullable(parseFailuresFromOperationsStates(response.getOperationStates(), response.getFailureReason()))
-        );
+                DatalakeOperationStatus.State.valueOf(response.getOverallState()),
+                List.of(), "", parseFailuresFromOperationsStates(response.getOperationStates(), response.getFailureReason()));
     }
 
     public DatalakeRestoreStatusResponse convert(RestoreDatalakeResponse response) {
         return new DatalakeRestoreStatusResponse(response.getBackupId(), response.getRestoreId(),
                 DatalakeRestoreStatusResponse.State.valueOf(response.getOverallState()),
-                Optional.ofNullable(parseFailuresFromOperationsStates(response.getOperationStates(), response.getFailureReason()))
+                parseFailuresFromOperationsStates(response.getOperationStates(), response.getFailureReason())
         );
     }
 
     public DatalakeBackupStatusResponse convert(BackupDatalakeStatusResponse response) {
+        List<String> contents = new LinkedList<>();
+        InternalBackupRestoreState operationStates = response.getOperationStates();
+        SolrBackupRestoreState solrState = operationStates.getSolr();
+        HbaseBackupRestoreState hbaseState = operationStates.getHbase();
+
+        if (solrState.getRangerAuditsCollection().getStatus().equals(SUCCESSFUL_STATUS)) {
```

Comment:
dontt we need to handle any null here?

Reply:
these (all the chain of objects) are from Protobuf generated definitions - they are all non-null